src/SendGrid/SendGridClient.cs

```diff
@@ -102,6 +112,21 @@ public SendGridClient(string apiKey, string host = null, Dictionary<string, stri
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SendGridClient"/> class.
+        /// </summary>
+        /// <param name="httpClient">The HTTP Client used to send requests to the SendGrid API.</param>
+        /// <param name="options">An <see cref="IOptions{SendGridClientOptions}"/> instance specifying the configuration to be used with the client.</param>
+        public SendGridClient(HttpClient httpClient, IOptions<SendGridClientOptions> options)
+        {
+            if (options?.Value == null)
```

Comment:
If this `if` is the reason you need to introduce `InitializeClient` you could use `this(httpClient, options?.Value ?? throw new ArgumentNullException(nameof(options)))`

Reply:
Thanks, I didn't know it could be expressed like that! I've refactored the code to include your suggestion.